Save flash: Replace CLI params for OSD stats and warnings with bitmasks

limonspb opened the pull request:
I think nobody uses CLI to set OSD stats or warning flags anyways.
Replacing them with a single bitmask saves:
1364B for F7X2
1352B for F411

blckmn commented:
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> FAIL
 - cooling off period lapsed -> FAIL
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> FAIL
 - `Tested` label found -> FAIL
 - assigned to an approver -> FAIL
 - approver count at least three -> FAIL

limonspb commented:
@daleckystepan why it needs coordination with Configurator? i think im missing something.

daleckystepan commented:
My bad. I forget that Configurator uses MSP not CLI directly.

limonspb commented:
@daleckystepan  could you please remove the tag "coordination". I don't have rights

daleckystepan commented:
@limonspb Removed

pfeerick commented:
So what is the expected upgrade mechanism/path for osd_stats from 4.3 => 4.4 since this means you can't use your previous diffs?

limonspb commented:
> So what is the expected upgrade mechanism/path for `osd_stats` from 4.3 => 4.4 since this means you can't use your previous diffs?

Between versions you can't use DIFF either way, it's pretty common.
In this case you will just have default OSD stats screen

pfeerick commented:
> Between versions you can't use DIFF either way, it's pretty common.

Er... sorry, but no. It's common that you can't just use the diff as is. But you can as long as you take care to omit incompatible settings. i.e. this is how I upgrade my quads from 4.2 => 4.3, with the help of the [Betaflight 4.3 CLI Migrator](https://bf43-safe-migration.robertodedomenico.it/)

So are you saying *there is no migration path for these settings* ... and while a minor annoyance, means it would be nice to have in the release notes mention that "OSD Warnings and Post Flight Statistics settings will need to be manually re-applied" or words to that effect?

limonspb commented:
> > Between versions you can't use DIFF either way, it's pretty common.
> 
> Er... sorry, but no. It's common that you can't just use the diff as is. But you can as long as you take care to omit incompatible settings. i.e. this is how I upgraded my quads from 4.2 => 4.3, with the help of the [Betaflight 4.3 CLI Migrator](https://bf43-safe-migration.robertodedomenico.it/)
> 
> So are you saying _there is no migration path for these settings_ ... and while a minor annoyance, means it would be nice to have in the release notes mention that "OSD Warnings and Post Flight Statistics settings will need to be manually re-applied" or words to that effect?

Between 4.2 and 4.3 there are BUNCH of parameters renamed, removed, replaced, and added. Just like this change. CLI migrator is a third party tool. Just like they are migrating 4.2 -> 4.3 they can make a migration 4.3 -> 4.4 if they want, this change is not preventing that.

And i will repeat again, DIFF from 4.2 is never fully compatible with 4.3. THere are some commands that are compatitable, but there are a lot that are not.

limonspb commented:
> > Between versions you can't use DIFF either way, it's pretty common.
> 
> Er... sorry, but no. It's common that you can't just use the diff as is. But you can as long as you take care to omit incompatible settings. i.e. this is how I upgraded my quads from 4.2 => 4.3, with the help of the [Betaflight 4.3 CLI Migrator](https://bf43-safe-migration.robertodedomenico.it/)
> 
> So are you saying _there is no migration path for these settings_ ... and while a minor annoyance, means it would be nice to have in the release notes mention that "OSD Warnings and Post Flight Statistics settings will need to be manually re-applied" or words to that effect?

from the release notes:
![image](https://user-images.githubusercontent.com/2925027/185086363-1316a97d-18f1-4c56-90eb-b3754fcfba14.png)

Third-party migration tools are cool, but Betaflight is not in control of them and obviously, they can't migrate everything.

sugaarK commented:
> > Between versions you can't use DIFF either way, it's pretty common.
> 
> Er... sorry, but no. It's common that you can't just use the diff as is. But you can as long as you take care to omit incompatible settings. i.e. this is how I upgraded my quads from 4.2 => 4.3, with the help of the [Betaflight 4.3 CLI Migrator](https://bf43-safe-migration.robertodedomenico.it/)
> 
> So are you saying _there is no migration path for these settings_ ... and while a minor annoyance, means it would be nice to have in the release notes mention that "OSD Warnings and Post Flight Statistics settings will need to be manually re-applied" or words to that effect?

You or some one else is welcome to Maintain such a project.. it’s not somthing we at Betaflight would put effort into though as we have much more pressing stuff to spend our free time on
